app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
```

Comment:
couldn't the caller of save emoji do a quick compare (hash compare if emojis are stored in a hash table) with the current list of emoji to figure out if Time needs to be restarted? additionally if this function is being called often maybe we need to rethink that. If it is not called that often then you could just restart IME without the check, but according to your reply sounds like this function is called often.